vm/map.c

```diff
@@ -1120,6 +1122,32 @@ int vm_mapBelongs(const struct _process_t *proc, const void *ptr, size_t size)
 }
 
 
+int vm_mapStringBelongs(const struct _process_t *proc, const char *str)
```

Comment:
Shouldn't there be a fast return for `str == NULL` or at least handle that per-syscall? To avoid arithmetic on `NULL`s and unnecessary locks.